cron/job.go

```diff
@@ -0,0 +1,357 @@
+}
+
+// MarshalFlatbuffer writes a job to a builder
+func (job *Job) MarshalFlatbuffer(builder *flatbuffers.Builder) flatbuffers.UOffsetT {
+	name := builder.CreateString(job.Name)
+	alias := builder.CreateString(job.Alias)
+
+	lastRunStart := builder.CreateString(job.LastRunStart.Format(time.RFC3339))
+	lastRunStop := builder.CreateString(job.LastRunStop.Format(time.RFC3339))
+	lastError := builder.CreateString(job.LastError)
+	logPath := builder.CreateString(job.LogFilePath)
+	p := builder.CreateString(job.Periodicity.String())
+
+	var opts flatbuffers.UOffsetT
+	if job.Options != nil {
+		opts = job.Options.MarshalFlatbuffer(builder)
+	}
+
+	cronfb.JobStart(builder)
+	cronfb.JobAddName(builder, name)
+	cronfb.JobAddAlias(builder, alias)
+
+	cronfb.JobAddType(builder, job.Type.Enum())
+	cronfb.JobAddPeriodicity(builder, p)
+	cronfb.JobAddLastRunStart(builder, lastRunStart)
+	cronfb.JobAddLastRunStop(builder, lastRunStop)
+	cronfb.JobAddLastError(builder, lastError)
+	cronfb.JobAddLogFilePath(builder, logPath)
+	cronfb.JobAddOptionsType(builder, job.fbOptionsType())
+	if opts != 0 {
+		cronfb.JobAddOptions(builder, opts)
+	}
+	return cronfb.JobEnd(builder)
+}
+
+func (job *Job) fbOptionsType() byte {
+	switch job.Options.(type) {
+	case *DatasetOptions:
+		return cronfb.OptionsDatasetOptions
+	default:
+		return 0 // will fire for nil case
+	}
+}
+
+// UnmarshalFlatbuffer decodes a job from a flatbuffer
+func (job *Job) UnmarshalFlatbuffer(j *cronfb.Job) error {
+	lastRunStart, err := time.Parse(time.RFC3339, string(j.LastRunStart()))
+	if err != nil {
+		return err
+	}
+
+	lastRunStop, err := time.Parse(time.RFC3339, string(j.LastRunStop()))
+	if err != nil {
+		return err
+	}
+
+	p, err := iso8601.ParseRepeatingInterval(string(j.Periodicity()))
+	if err != nil {
+		return err
+	}
+
+	*job = Job{
+		Name:        string(j.Name()),
+		Alias:       string(j.Alias()),
+		Type:        JobType(cronfb.EnumNamesJobType[j.Type()]),
+		Periodicity: p,
+
+		LastRunStart: lastRunStart,
+		LastRunStop:  lastRunStop,
+		LastError:    string(j.LastError()),
+		LogFilePath:  string(j.LogFilePath()),
+	}
+
+	unionTable := new(flatbuffers.Table)
+	if j.Options(unionTable) {
+		if j.OptionsType() == cronfb.OptionsDatasetOptions {
+			fbOpts := &cronfb.DatasetOptions{}
+			fbOpts.Init(unionTable.Bytes, unionTable.Pos)
+			opts := &DatasetOptions{}
+			opts.UnmarshalFlatbuffer(fbOpts)
+			job.Options = opts
+		}
+	}
+
+	return nil
+}
+
+// Options is an interface for job options
+type Options interface {
+	MarshalFlatbuffer(builder *flatbuffers.Builder) flatbuffers.UOffsetT
+}
+
+// ShellScriptOptions encapsulates options for running a shell script cron job
+type ShellScriptOptions struct {
+	// none yet.
+}
+
+// MarshalFlatbuffer writes to a builder
+func (o *ShellScriptOptions) MarshalFlatbuffer(builder *flatbuffers.Builder) flatbuffers.UOffsetT {
+	return 0
+}
+
+// DatasetOptions encapsulates options passed to `qri save`
+type DatasetOptions struct {
```

Comment:
Yeah I think long-term the answer is to use a flexbuffer here & remove the concrete options implementations from Cron completely (because no schema will be required). But we'll need to contributed golang support for flexbuffers first.